tensorflow_datasets/image/stanford_dogs.py

```diff
@@ -0,0 +1,176 @@
+    VERSION = tfds.core.Version("0.1.0")
+    
+    def _info(self):
+
+        return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            # Images are of varying size
+            "image": tfds.features.Image(),
+            "image/filename": tfds.features.Text(),
+            "label": tfds.features.ClassLabel(num_classes=120),
+            # Multiple bounding box per image
+            "objects": tfds.features.SequenceDict({
+                "bbox": tfds.features.BBoxFeature(),
+            }),
+        }),
+        supervised_keys=("image", "label"),
+        urls=_URL,
+        citation=_CITATION
+        )
+
+    def _split_generators(self, dl_manager):
+
+        download_path = dl_manager.download([_SPLIT_URL, _ANNOTATIONS_URL, _IMAGES_URL])
+        extracted_path = dl_manager.download_and_extract([_SPLIT_URL, _ANNOTATIONS_URL])
+        xml_file_list = defaultdict(str)
+
+        def parse_mat_file(file_name):
+
+            parsed_mat_arr = tfds.core.lazy_imports.scipy_io.loadmat(file_name, squeeze_me=True)
+            file_list = [element.split('/')[-1] for element in parsed_mat_arr['file_list']]
+
+            return file_list, parsed_mat_arr
+
+        for fname in tf.io.gfile.listdir(extracted_path[0]):
+            # Train-test split using train_list.mat and test_list.mat
+            full_file_name = os.path.join(extracted_path[0], fname)
+
+            if "train" in fname: 
+                train_list, train_mat_arr = parse_mat_file(full_file_name)
+                label_names = set([element.split('/')[-2].lower() for element in train_mat_arr['file_list']])
+            elif "test" in fname:
+                test_list, _ = parse_mat_file(full_file_name)
+
+        self.info.features["label"].names = tuple(label_names)
+        
+        for root, _, files in tf.io.gfile.walk(extracted_path[1]):
+            # Parsing the XML file which have the image annotations
+            for fname in files:
+                annotation_file_name = os.path.join(root, fname)
+                xml_file_list[fname] = ET.parse(annotation_file_name)
+
+        
+        return [
+            tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=10,
+            gen_kwargs={
+            "archive": dl_manager.iter_archive(download_path[2]),
+            "file_names": train_list,
+            "annotation_files": xml_file_list,
+            }),
+            
+            tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            num_shards=1,
+            gen_kwargs={
+            "archive": dl_manager.iter_archive(download_path[2]),
+            "file_names": test_list,
+            "annotation_files": xml_file_list,
+            })
+        ]
+        
+    
+    def _generate_examples(self, archive, file_names, annotation_files):
+        """Generate horses or humans images and labels given the directory path

```

Comment:
Kindly change the docstring to the dataset name you are adding :P

Reply:
This has already been done. You are looking at an outdated file.